src/sqlfluff/rules/L049.py

```diff
@@ -41,79 +42,60 @@ class Rule_L049(Rule_L006):
     """
 
     groups = ("all", "core")
-    # Inherit crawl behaviour from L006
+    crawl_behaviour = SegmentSeekerCrawler({"comparison_operator"})
 
-    def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Relational operators should not be used to check for NULL values."""
         # Context/motivation for this rule:
         # https://news.ycombinator.com/item?id=28772289
         # https://stackoverflow.com/questions/9581745/sql-is-null-and-null
-        if len(context.segment.segments) <= 2:
-            return None  # pragma: no cover
 
         # Allow assignments in SET clauses
-        if context.parent_stack and context.parent_stack[-1].is_type(
+        if context.parent_stack and context.parent_stack[-2].is_type(
```

Comment:
Is there a risk that `parent_stack[-2]` raises an `IndexError` (i.e. if there are less than 2 ancestors in the stack)?

Reply:
good point - I'll have a look.